pandas/core/dtypes/dtypes.py

```diff
@@ -126,16 +131,20 @@ class PandasExtensionDtype(_DtypeOpsMixin):
     isnative = 0
     _cache = {}  # type: Dict[str_type, 'PandasExtensionDtype']
 
-    def __unicode__(self):
+    def __unicode__(self) -> str_type:
         return self.name
 
-    def __str__(self):
+    @property
+    def name(self) -> str_type:
+        raise AbstractMethodError(self)
```

Comment:
why are you using str_type? just str

Reply:
I think `str_type` came about in a previous PR because the class has a variable also called `str` which was conflicting with the builtin. 

@gwrome

Reply:
that is not a good idea at all. pls revert that

Reply:
So the other option that was proposed was to namespace it as builtins.str within the scope of the class - do you prefer that?

Reply:
The discussions about str getting shadowed were in #26029, for reference. I still prefer the explicit builtins.str approach.